lib/resourcemerge/core.go

```diff
@@ -50,11 +63,12 @@ func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.Pod
 		}
 		if existingCurr == nil {
 			*modified = true
-			existing.Containers = append(existing.Containers, corev1.Container{})
-			existingCurr = &existing.Containers[len(existing.Containers)-1]
+			existingCurr = &corev1.Container{}
```

Comment:
So if I'm interpreting this right, this is "if there's one in the old list with the same name as the new list, replace it with an empty object so the new object's fields always "win" the merge" ? 

The new ones always make it in, the old ones that match get blanked to be overwritten, and the old ones that don't match never make it into the list. 

That seems like we eventually get the outcome we want, but you think this is buying us anything over just  replacing the list if it's different, e.g.: 
```
if !equality.Semantic.DeepEqual(required.Containers, *existing.Containers) {
		*existing.Containers = required.Containers
		*modified = true
}
```

Reply:
I think this may work @jkyros. I am trying to think of scenarios where ensureContainer doesn't just take required (the new ctr) but i can't seem to find one

the way we do things now in this PR:

OK, for each required container, if an existing ctr exists with the same name, compare them and merge the entities but really just take everything from required and get rid of existing.

the deep equal way:

OK, if the new set of containers differs from the old one, take the entire new set.

Seems like the same outcome to me, lets try it.